pkg/api/customization/globaldns/action_globaldns.go

```diff
@@ -118,12 +120,37 @@ func (w *Wrapper) removeProjects(gDNS *v3.GlobalDNS, request *types.APIContext,
 			finalProjects = append(finalProjects, p)
 		}
 	}
-	gDNSToUpdate.Spec.ProjectNames = finalProjects
-	return w.updateGDNS(gDNSToUpdate, request, "removedProjects")
+	return w.updateGDNS(gDNS, finalProjects, request, "removedProjects")
 }
 
-func (w Wrapper) updateGDNS(gDNSToUpdate *v3.GlobalDNS, request *types.APIContext, message string) error {
-	if _, err := w.GlobalDNSes.Update(gDNSToUpdate); err != nil {
+func (w Wrapper) updateGDNS(gDNS *v3.GlobalDNS, targetProjects []string, request *types.APIContext, message string) error {
+	backoff := wait.Backoff{
+		Duration: 100 * time.Millisecond,
+		Factor:   2,
+		Jitter:   0.5,
+		Steps:    7,
+	}
+
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		gDNSlatest, err := w.GlobalDNSes.GetNamespaced(gDNS.Namespace, gDNS.Name, v1.GetOptions{})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return false, err
+			}
+			return false, nil
```

Comment:
Why do we return nil if object is not found? If this is just called by action handler, maybe we can just return the msg?

Reply:
Yeah I guess that makes sense. I borrowed this from an existing piece of code using ExponentialBackoff, but it the action handler can't find the obj on Get then we can return that error